### DISCUSSION TITLE
Completely Clear A Scene

### DISCUSSION THREAD

**Discussion by N8python** [enable3d/enable3d#114] [Help / Questions] [answered]
Is there any function that would destroy all objects, including the ground, within the scene?
So that I could use `warpSpeed` again.

**Comment by yandeu** [comment, marked as answer]
If you call `this.restart()` inside your `Scene3D`:
- all elements are destroyed
- Scene3D will again call `init()`, `preload()` and `create()`
- the internal cache will not be emptied
- since the cache remains, the browser won't load your assets again.

`this.start()` is similar, but it starts another scene instead.

See this example:
https://enable3d.io/examples/project-options-and-multiple-scenes.html

---

If you use enable3d it with **Phaser** you can simply use Phaser's SceneManager.

See this example:
https://enable3d.io/examples/texture-cube.html

**Reply by N8python** [reply]
Awesome, thanks so much!

**Reply by N8python** [reply]
Sorry to trouble you, but for some reason the restart method isn't defined on my scene. Not sure if I'm using an outdated version or something. Is it because I'm using phaser?

**Reply by yandeu** [reply]
> Is it because I'm using phaser?

Yes. I have just updated my answer.

**Reply by N8python** [reply]
Epic, thx.
